vispTypes.hpp

```diff
@@ -38,6 +39,23 @@ namespace visp
         float gain_at_infinity;
         float slope_at_zero;
     };
+
+    /** Group relevant information about the controller
+     *
+     *  error        - the norm-2 of the error vector
+     *  residual     - value expressed in meter that represents the
+                       "quality" of the measurement"
+     *  desired_pose - the current setpoint
+     *  current_pose - the current pose
+     */
+    struct controllerState
+    {
+        double error;
```

Comment:
"Logical". If you get an image that's 150ms late, then your velocity is the velocity you should have generated 150ms ago. Not now.

Recommended reading: http://rock-robotics.org/master/documentation/data_processing/data.html